app/models/service.rb

```diff
@@ -188,7 +189,7 @@ def self.all
     end
 
     event :mark_as_deleted do
-      transition [:incomplete, :published, :offline, :hidden] => :deleted, unless: :last_accessible?
+      transition [:incomplete, :published, :offline, :hidden] => :deleted, unless: :default_or_last?
```

Comment:
Why changing this?
We should be able to delete the default service

Reply:
OK got my answer:

> The second commit makes the condition to mark a service as deleted to be the same as the condition to actually destroy it.

Reply:
Exactly 😄Right now in master we allow to mark as deleted unless `last_accessible?`, but we allow to destroy for real unless `default_or_last?`, which means that if it is the default, it can be marked as deleted but it can not be destroyed for real.

#### Validation for marking as deleted
https://github.com/3scale/porta/blob/957fe61e7611cdb9e28a07a4ccc75530a6004db2/app/models/service.rb#L191

#### Validation for destroying the record for real
https://github.com/3scale/porta/blob/957fe61e7611cdb9e28a07a4ccc75530a6004db2/app/models/service.rb#L273-L277
https://github.com/3scale/porta/blob/957fe61e7611cdb9e28a07a4ccc75530a6004db2/app/models/service.rb#L512-L515
https://github.com/3scale/porta/blob/957fe61e7611cdb9e28a07a4ccc75530a6004db2/app/models/service.rb#L449-L451